src/assocPath.js

```diff
@@ -41,7 +43,8 @@ module.exports = _curry3(function assocPath(path, val, obj) {
   }
   if (_isInteger(idx) && _isArray(obj)) {
     var arr = [].concat(obj);
-    arr[idx] = val;
+    var arrIdx = idx < 0 ? arr.length + idx : idx;
```

Comment:
Sorry for the delay on this one @keithbro . I'd prefer `assoc` to be `function assoc(prop, val, obj) { return assocPath([prop], val, obj); }` and for all the logic to be in `assocPath`. But let's see what others think.. anyone?